Framework/CurveFitting/src/FuncMinimizers/FABADAMinimizer.cpp

```diff
@@ -976,53 +650,482 @@ void FABADAMinimizer::SimAnnealingRefrigeration() {
+    auto &X = wsConv->mutableX(j);
+    auto &Y = wsConv->mutableY(j);
+    for (size_t k = 0; k < convLength; ++k) {
+      X[k] = double(k);
+      Y[k] = conv_chain[nSteps * k];
+    }
+  }
+
+  // Set and name the workspace for the converged part of the chain.
+  setProperty("ConvergedChain", wsConv);
+}
+
+/** Create the workspace containing chi2 values
+*
+* @param convLength :: length of the converged chain
+* @param mostProbableChi2 :: most probable chi2 value
+*correlation
+*/
+void FABADAMinimizer::outputCostFunctionTable(size_t convLength,
+                                              double mostProbableChi2) {
+  // Create the workspace for the Chi square values.
+  API::ITableWorkspace_sptr wsChi2 =
+      API::WorkspaceFactory::Instance().createTable("TableWorkspace");
+  wsChi2->addColumn("double", "Chi2min");
+  wsChi2->addColumn("double", "Chi2MP");
+  wsChi2->addColumn("double", "Chi2min_red");
+  wsChi2->addColumn("double", "Chi2MP_red");
+
+  // Obtain the quantity of the initial data.
+  API::FunctionDomain_sptr domain = m_leastSquares->getDomain();
+  size_t data_number = domain->size();
+
+  // Calculate the value for the reduced Chi square.
+  double Chi2min_red =
+      m_chi2 / (double(data_number - m_nParams)); // For de minimum value.
+  double mostPchi2_red;
+  if (convLength > 0)
+    mostPchi2_red = mostProbableChi2 / (double(data_number - m_nParams));
+  else {
+    g_log.warning()
+        << "Most probable chi square not calculated. -1 is returned.\n"
+        << "Most probable reduced chi square not calculated. -1 is "
+           "returned.\n";
+    mostPchi2_red = -1;
+  }
+
+  // Add the information to the workspace and name it.
+  API::TableRow row = wsChi2->appendRow();
+  row << m_chi2 << mostProbableChi2 << Chi2min_red << mostPchi2_red;
+  setProperty("CostFunctionTable", wsChi2);
+}
+
+/** Create the workspace containing PDF
+*
+* @param convLength :: length of the converged chain
+* @param reducedChain :: the reduced chain (will be sorted)
+* @return :: most probable chi square value
+*/
+double
+FABADAMinimizer::outputPDF(size_t convLength,
+                           std::vector<std::vector<double>> &reducedChain) {
+
+  // To store the most probable chi square value
+  double mostPchi2;
+
+  // Create the workspace for the Probability Density Functions
+  int pdfLength = getProperty(
+      "NumberBinsPDF"); // histogram length for the PDF output workspace
+  if (pdfLength <= 0) {
+    g_log.warning() << "Non valid Number of bins for the PDF (<= 0)."
+                       " Default value (20 bins) taken\n";
+    pdfLength = 20;
+  }
+  API::MatrixWorkspace_sptr ws = API::WorkspaceFactory::Instance().create(
+      "Workspace2D", m_nParams + 1, pdfLength + 1, pdfLength);
+
+  // Calculate the cost function Probability Density Function
+  if (convLength > 0) {
+    std::sort(reducedChain[m_nParams].begin(), reducedChain[m_nParams].end());
+    std::vector<double> pdf_y(pdfLength, 0);
```

Comment:
It is actually correct: pdf(chi^2|Data) := Integral over A of 1*dx
Where dx is the volume differential of the parameters and A = {x|pdf(x|Data) = exp(-chi^2/2)}